app/modules/net.c

```diff
@@ -202,20 +202,26 @@ static void net_recv_cb(lnet_userdata *ud, struct pbuf *p, ip_addr_t *addr, u16_
     pbuf_free(p);
     return;
   }
+
   lua_State *L = lua_getstate();
-  int num_args = 2;
-  lua_rawgeti(L, LUA_REGISTRYINDEX, ud->client.cb_receive_ref);
-  lua_rawgeti(L, LUA_REGISTRYINDEX, ud->self_ref);
-  lua_pushlstring(L, p->payload, p->len);
-  if (ud->type == TYPE_UDP_SOCKET) {
-    num_args += 2;
-    char iptmp[16];
-    bzero(iptmp, 16);
-    ets_sprintf(iptmp, IPSTR, IP2STR(&addr->addr));
-    lua_pushinteger(L, port);
-    lua_pushstring(L, iptmp);
+  struct pbuf *pp = p;
+  while (pp)
+  {
+    int num_args = 2;
+    lua_rawgeti(L, LUA_REGISTRYINDEX, ud->client.cb_receive_ref);
+    lua_rawgeti(L, LUA_REGISTRYINDEX, ud->self_ref);
+    lua_pushlstring(L, pp->payload, pp->len);
+    if (ud->type == TYPE_UDP_SOCKET) {
+      num_args += 2;
+      char iptmp[16];
+      bzero(iptmp, 16);
+      ets_sprintf(iptmp, IPSTR, IP2STR(&addr->addr));
```

Comment:
Can be moved outside a loop

Reply:
Yes, though it becomes (slightly) harder to follow, and you double up on the `if` statement as you'll need to push the port/ip each time through the loop regardless. Given that we'd normally be doing the `ets_sprintf` for each pbuf anyway, I felt the tradeoff wasn't worth the readability cost. If you really reckon it is, I can change it.

Reply:
Agree for "harder to follow"...
Will accept 'FIXME' comment there as a solution for now.